src/hardhat/contracts/Staking/FraxUnifiedFarm_ERC20.sol

```diff
@@ -372,25 +488,41 @@ contract FraxUnifiedFarm_ERC20 is FraxUnifiedFarmTemplate {
     // Two different withdrawLocked functions are needed because of delegateCall and msg.sender issues (important for migration)
     function withdrawLocked(bytes32 kek_id, address destination_address) nonReentrant external returns (uint256) {
         require(withdrawalsPaused == false, "Withdrawals paused");
-        return _withdrawLocked(msg.sender, destination_address, kek_id);
+
+        // Backwards compat - lookup the index of the stake item first.
+        uint256 arr_idx = getLockedStakeArrIndex(msg.sender, kek_id);
+        return _withdrawLocked(msg.sender, destination_address, kek_id, arr_idx);
+    }
+
+    // Overloaded to take the arr_idx such that it can be computed off-chain.
```

Comment:
Is computing the index off chain reasonable?

Is it not that the case that the array index calculated may become out of date before it is used in which case _getState () will fail with "Stake not found" ?

Reply:
Yes it could given we're now `pop()`'ing the item out of the array (leaving as a delete wouldn't change the arr_idx)

It was my concern too, but according to @buterlji it's ok to make it a concern of the client to handle correctly.

That's one of the reasons I left the old version too which only takes the kek_id...again up to FRAX to comment operationally I think.

Reply:
> make it a concern of the client to handle correctly.

@butlerji  I don't see how this would work. Would the client specifically detect the "Stake not found" error, and then know this is a reason to retry?

Reply:
I think the flow would be pretty synchronous and single-threaded for a given address. 

So first start with: `myLockedStakes = contract.lockedStakesOf(myAddress)`

Then if the client calls `withdrawLocked()` then its local array should change too for any subsequent calls (or just call `contract.lockedStakesOf(myAddress)` again)

Reply:
> I think the flow would be pretty synchronous and single-threaded for a given address.

Oh - I misunderstood. I thought the array was being used for all addresses.